account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -36,12 +68,49 @@ def action_account_change_currency(self):
                 line.price_unit *= rate
             for tax in invoice.tax_line_ids:
                 tax.amount *= rate
+            invoice.date_invoice = today
```

Comment:
Re-assign the date from the top of the method just if the date_invoice is empty
and remove the `invoice.date_invoice or today` since that even should be filled if this button is used.